kroxylicious-api/src/main/java/io/kroxylicious/proxy/filter/KrpcFilterContext.java

```diff
@@ -93,4 +93,9 @@ public interface KrpcFilterContext {
     // void closeConnection();
 
     // TODO an API to allow a filter to add/remove another filter from the pipeline
+
+    static <T extends ApiMessage> KrpcFilterContext<T> toSpecificContext(KrpcFilterContext<ApiMessage> context) {
+        // TODO find less ugly way? You can alternatively make a wrapper KrpcFilterContext
```

Comment:
I wondered about creating api specific context implementations too. FilterHander would need something to create the correct implementation.  And we'd need a way to handle the RequestFilter/ResponseFilter cases.

I felt like I was digging a big horrid hole.